src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/ReadMePreviewViewModel.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.VisualStudio.Telemetry;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public sealed class ReadMePreviewViewModel : ViewModelBase
+    {
+        public ReadMePreviewViewModel()
+        {
+            _isErrorWithReadMe = false;
+            _rawReadMe = string.Empty;
+        }
+
+        private bool _isErrorWithReadMe = false;
+
+        public bool IsErrorWithReadMe
+        {
+            get => _isErrorWithReadMe;
+            set
+            {
+                if (_isErrorWithReadMe != value)
+                {
+                    _isErrorWithReadMe = value;
+                    RaisePropertyChanged(nameof(IsErrorWithReadMe));
+                }
+            }
+        }
+
+        private string _rawReadMe = string.Empty;
+
+        public string ReadMeMarkdown
+        {
+            get => _rawReadMe;
+            set
+            {
+                if (_rawReadMe != value)
+                {
+                    _rawReadMe = value;
+                    RaisePropertyChanged(nameof(ReadMeMarkdown));
+                }
+            }
+        }
+
+        private bool _canDetermineReadMeDefined = true;
+
+        public bool CanDetermineReadMeDefined
+        {
+            get => _canDetermineReadMeDefined;
+            set
+            {
+                if (_canDetermineReadMeDefined != value)
+                {
+                    _canDetermineReadMeDefined = value;
+                    RaisePropertyChanged(nameof(CanDetermineReadMeDefined));
+                }
+            }
+        }
+
+        public async Task LoadReadme(string packagePath, string packageId)
+        {
+            var newReadMeValue = string.Empty;
+            var isErrorWithReadMe = false;
+            bool canDetermineReadMeDefined = false;
+#pragma warning disable CA1031 // Do not catch general exception types
+            try
+            {
+                if (!string.IsNullOrEmpty(packagePath))
+                {
+                    var packageDirectory = Path.GetDirectoryName(packagePath);
+                    var nuspecPath = Path.Combine(packageDirectory, $"{packageId}{PackagingCoreConstants.NuspecExtension}");
+                    var nuspectFileInfo = new FileInfo(nuspecPath);
```

Comment:
Maybe NuGet.Protocol's SourceRepository can be used to abstract away local vs http feeds